src/cpu/aarch64/matmul/brgemm_matmul_utils.hpp

```diff
@@ -0,0 +1,324 @@
+
+    int first_bcast_dim;
+    int last_bcast_dim;
+    bool bcast_across_all_batch_dims;
+
+    dim_t first_bcast_dim_to_last_batch_dim_prod;
+    dim_t bcast_dims_prod;
+
+    dim_t batch_dims[max_batch_ndims];
+    dim_t gb_off[max_batch_ndims]; // generalized batch offset
+};
+
+struct brgemm_matmul_conf_t {
+    int ndims, batch_ndims;
+    dim_t M, N, K, batch, batch_without_first_dim;
+    dim_t M_blk, N_blk, K_blk, M_tail, N_tail, K_tail;
+    int M_chunk_size, N_chunk_size;
+    dim_t LDA, LDB, LDC, LDD;
+    int brgemm_batch_size, brgemm_batch_tail_size;
+    int wei_n_blk, wei_k_blk;
+    brgemm_batch_kind_t brg_type;
+
+    cpu_isa_t isa;
+
+    format_tag_t src_tag, wei_tag, dst_tag, bia_tag;
+    bool with_bias;
+    bool with_sum;
+    bool with_eltwise;
+    bool with_binary;
+    bool with_scales;
+    bool with_dst_scales;
+    bool s8s8_compensation_required;
+    bool is_oscale_per_n;
+    brgemm_broadcast_t src_zp_type;
+    brgemm_broadcast_t wei_zp_type;
+    brgemm_broadcast_t dst_zp_type;
+
+    bool use_buffer_a;
+    bool use_buffer_a_tail_only;
+    bool use_buffer_b;
+    bool use_buffer_c;
+
+    brgemm_matmul_bcast_desc_t bcast_A_desc;
+    brgemm_matmul_bcast_desc_t bcast_B_desc;
+
+    data_type_t src_dt;
+    data_type_t dst_dt;
+    data_type_t wei_dt;
+    data_type_t acc_dt;
+    data_type_t bia_dt;
+    int nthr;
+    int nthr_k;
+
+    // Auxiliary values for init_config() and execute()
+    dim_t a_dt_sz, b_dt_sz, c_dt_sz, acc_dt_sz, bias_dt_sz;
+
+    // used for transposed buffer datatype when different from x_dt_sz
+    // (e.g. used in BF32 implementations having to down-convert to BF16
+    // from FP32 implementation)
+    dim_t tr_a_dt_sz, tr_b_dt_sz;
+
+    int M_chunks;
+    int N_chunks;
+    int K_chunks;
+    int num_M_blocks;
+    int num_N_blocks;
+    dim_t M_chunk_elems;
+    dim_t N_chunk_elems;
+    dim_t K_chunk_elems;
+
+    // Pre-calculated memory strides for each tensor
+    dim_t A_strides[3];
+    dim_t B_strides[3];
+    dim_t C_strides[3];
+    dim_t buffer_c_chunk_sz;
+    dim_t buffer_c_per_thread_sz;
+
+    dim_t A_ptr_shift_b;
+    dim_t B_ptr_shift_b;
+    dim_t C_ptr_shift_b;
+    dim_t copy_A_src_stride;
+    dim_t copy_B_wei_stride;
+
+    dim_t buffer_a_chunk_sz;
+    dim_t buffer_a_chunk_shift_along_m;
+    dim_t buffer_a_per_thread_sz;
+
+    dim_t buffer_b_chunk_sz;
+    dim_t buffer_b_per_thread_sz;
+    dim_t s8s8_comp_ithr_str;
+    dim_t s8s8_comp_b_str;
+    dim_t s8s8_comp_n_str;
+    bool has_zero_point_a, has_zero_point_b, has_zero_point_c;
+    bool post_ops_applicable;
+    bool transposed_A;
+    bool blocked_B;
+
+    dim_t zp_a_comp_shift_n;
+    dim_t zp_a_comp_elems_per_thr;
+
+    dim_t zp_b_comp_result_shift_m;
+    dim_t zp_b_comp_buffer_start;
+    dim_t zp_b_comp_buffer_shift_m;
+    dim_t zp_b_comp_elems_per_thr;
+
+    int wsp_tile_per_thr_bytes;
+    int brgemm_batch_element_per_thr_sz;
+    bool is_amx;
```

Comment:
The `AMX` does not make sense for Arm CPUs. A possible alternative on Arm side is `SME` extension, but since it is not supported there are a lot of code that is "dead", because there is no support for `SME`. I would recommend to remove it as it will make the code simpler.